ext/storage_sources/gcp_store/gcp_storage_source.cpp

```diff
@@ -291,89 +310,145 @@ gcp_file_open(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name
-    WT_UNUSED(from);
-    WT_UNUSED(to);
-    WT_UNUSED(flags);
+    std::cerr << "gcp_rename: file renaming is not supported." << std::endl;
+    return ENOTSUP;
+}
+
+static int
+gcp_object_size(WT_FILE_SYSTEM *file_system, [[maybe_unused]] WT_SESSION *session, const char *name,
+  wt_off_t *sizep)
+{
+    bool exists;
+    size_t size;
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    int ret = fs->gcp_conn->object_exists(name, exists, size);
+    if (ret != 0) {
+        std::cerr << "gcp_object_size: GetObjectMetadata request to google cloud failed."
+                  << std::endl;
+        return (ret);
+    }
+    *sizep = size;
 
     return 0;
 }
 
+// Helper function to return a list of object names for the given location.
 static int
-gcp_file_size(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *name, wt_off_t *sizep)
+gcp_object_list_helper(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *directory,
+  const char *prefix, char ***object_list, uint32_t *count, bool list_single)
 {
-    WT_UNUSED(file_system);
-    WT_UNUSED(session);
-    WT_UNUSED(name);
-    WT_UNUSED(sizep);
+    gcp_file_system *fs = reinterpret_cast<gcp_file_system *>(file_system);
+    gcp_store *gcp = reinterpret_cast<gcp_file_system *>(fs)->storage_source;
+    std::vector<std::string> objects;
+    std::string complete_prefix;
 
-    return 0;
+    *count = 0;
+
+    if (directory != nullptr) {
+        complete_prefix += directory;
+        // Add a terminating '/' if one doesn't exist.
+        if (!complete_prefix.empty() && complete_prefix.back() != '/')
+            complete_prefix += '/';
+    }
+
+    if (prefix != nullptr)
+        complete_prefix += prefix;
+
+    int ret;
+
+    ret = list_single ? fs->gcp_conn->list_objects(complete_prefix, objects, true) :
+                        fs->gcp_conn->list_objects(complete_prefix, objects, false);
+
+    if (ret != 0) {
+        std::cerr << "gcp_object_list: ListObjects request to google cloud failed." << std::endl;
+        return (ret);
+    }
+    *count = objects.size();
+
+    std::cerr << "gcp_object_list: ListObjects request to google cloud succeeded. Received " +
+        std::to_string(*count) + " objects."
+              << std::endl;
+    gcp_object_list_add(*gcp, object_list, objects, *count);
+
+    return ret;
 }
 
+// Return a list of object names for the given location.
 static int
 gcp_object_list(WT_FILE_SYSTEM *file_system, WT_SESSION *session, const char *directory,
   const char *prefix, char ***object_list, uint32_t *count)
 {
-    WT_UNUSED(file_system);
-    WT_UNUSED(session);
-    WT_UNUSED(directory);
-    WT_UNUSED(prefix);
-    WT_UNUSED(object_list);
-    WT_UNUSED(count);
-
-    return 0;
+    return gcp_object_list_helper(
+      file_system, session, directory, prefix, object_list, count, false);
 }
 
+// Add objects retrieved from Google cloud bucket into the object list, and allocate the memory
+// needed.
```

Comment:
This comment and function name are discordant with the implementation. Apologies: I should have read more carefully before suggesting code.
The comment and funcname both imply that the c++ strings (objects) will be added to array of c strings (object_list).
The implementation only sets object_list iff return==0, it does not add anything.
Given the unit tests pass I'm assuming the implementation is correct, so perhaps a better name might be: `gcp_object_list_from_objects`.
It can also be helpful to explicit state changes to global state, specifically:
 * if return==0 and then object_list now points to allocated memory that is the responsibility of the caller to manage unless count==0 and then object_list==nullptr
  * if return!=0 object_list is unchanged